lib/webdriveragent.js

```diff
@@ -287,6 +288,76 @@ class WebDriverAgent {
     }
   }
 
+
+  /**
+   * @typedef {Object} LaunchWdaViaDeviceCtlOptions
+   * @property {Record<string, string|number>} [env] environment variables for the launching WDA process
+   */
+
+  /**
+   * Launch WDA with preinstalled package with 'xcrun devicectl device process launch'.
+   * The WDA package must be prepared properly like published via
+   * https://github.com/appium/WebDriverAgent/releases
+   * with proper sign for this case.
+   *
+   * When we implement launching XCTest service via appium-ios-device,
+   * this implementation can be replaced with it.
+   *
+   * @param {LaunchWdaViaDeviceCtlOptions} [opts={}] launching WDA with devicectl command options.
+   * @return {Promise<void>}
+   */
+
+  async _launchWdaViaDeviceCtl(opts = {}) {
+    // FIXME: use appium-xcuitest-driver's Devicectl. Maybe it needs to be included in the `this.device`?
+    //
+
+    const {env} = opts;
+
+    let xcrunBinnaryPath;
+    try {
+      xcrunBinnaryPath = await fs.which('xcrun');
+    } catch (e) {
+      throw new Error(
+        `xcrun has not been found in PATH. ` +
+          `Please make sure XCode development tools are installed`,
+      );
+    }
+
+    const cmd = [
+      'devicectl',
+      'device',
+      'process',
+      'launch',
+      `--device`, this.device.udid,
+      '--terminate-existing'
+    ];
+
```

Comment:
building `cmd` belongs to the same logic block. There is no need for extra line breaks